zboxcore/encryption/pre_test.go

```diff
@@ -0,0 +1,133 @@
+package encryption
+
+import (
+	"encoding/base64"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.dedis.ch/kyber/v3/group/edwards25519"
+	"testing"
+)
+
+func TestMnemonic(t *testing.T) {
+	mnemonic := "travel twenty hen negative fresh sentence hen flat swift embody increase juice eternal satisfy want vessel matter honey video begin dutch trigger romance assault"
+
+	encscheme := NewEncryptionScheme()
+
+	err := encscheme.Initialize(mnemonic)
+	require.NoError(t, err)
+
+
+	encscheme.InitForEncryption("filetype:audio")
+	pvk, _ := encscheme.GetPrivateKey()
+	expectedPvk := "XsQLPaRBOFS+3KfXq2/uyAPE+/qq3VW0OkW0T9q93wQ="
+	require.Equal(t, expectedPvk, pvk)
+	pubk, _ := encscheme.GetPublicKey()
+	expectedPubk := "PwpVIXgXbnt8NJmy+R4aSwG8HwJbsbT2JVQqa0bayZQ="
+	require.Equal(t, expectedPubk, pubk)
+
+}
+
+func TestReEncryptionAndDecryptionForShareData(t *testing.T) {
+	client_mnemonic := "travel twenty hen negative fresh sentence hen flat swift embody increase juice eternal satisfy want vessel matter honey video begin dutch trigger romance assault"
+	client_encscheme := NewEncryptionScheme()
+	client_encscheme.Initialize(client_mnemonic)
+	client_encscheme.InitForEncryption("filetype:audio")
+	client_enc_pub_key, err := client_encscheme.GetPublicKey()
+	assert.Nil(t, err)
+
+	shared_client_mnemonic := "inside february piece turkey offer merry select combine tissue wave wet shift room afraid december gown mean brick speak grant gain become toy clown"
+	shared_client_encscheme := NewEncryptionScheme()
+	shared_client_encscheme.Initialize(shared_client_mnemonic)
+	shared_client_encscheme.InitForEncryption("filetype:audio")
+
+	enc_msg, err := shared_client_encscheme.Encrypt([]byte("encrypted_data_uttam"))
+	assert.Nil(t, err)
+	regenkey, err := shared_client_encscheme.GetReGenKey(client_enc_pub_key, "filetype:audio")
+	assert.Nil(t, err)
+	enc_msg.ReEncryptionKey = regenkey
+
+	client_decryption_scheme := NewEncryptionScheme()
+	client_decryption_scheme.Initialize(client_mnemonic)
+	client_decryption_scheme.InitForDecryption("filetype:audio", enc_msg.EncryptedKey)
+
+	result, err := client_decryption_scheme.Decrypt(enc_msg)
+	assert.Nil(t, err)
+	assert.Equal(t, string(result), "encrypted_data_uttam")
+}
+
+func TestReEncryptionAndDecryptionForMarketplaceShare(t *testing.T) {
```

Comment:
Thanks for writing a unit test, but could you use the table driven subtests style https://blog.golang.org/subtests#TOC_4.